### PR TITLE
Import export sample as chemical

### PR DESCRIPTION
**Pull request by adambasha0**
- [ ] rather 1-story 1-commit than sub-atomic commits

- [x] commit title is meaningful =>  git history search

- [x] commit description is helpful => helps the reviewer to understand the changes

- [x] code is up-to-date with the latest developments of the target branch (rebased to it or whatever) => :fast_forward:-merge for linear history is favoured

- [x] added code is linted

- [ ] tests are passing (at least locally): we still have some random test failure on CI. thinking of asking spec/examples.txt to be commited

- [ ] in case the changes are visible to the end-user,  video or screenshots should be added to the PR => helps  with user testing

- [x] testing coverage improvement is improved.

- [ ] CHANGELOG :  add a bullet point on top (optional: reference to github issue/PR )

- [ ] parallele PR for documentation  on docusaurus  if the feature/fix is tagged for a release

### CODE REVIEW

File: app/packs/src/components/contextActions/ExportImportButton.js [hunk 1]
```diff
@@ -73,9 +97,8 @@ ExportImportButton.defaultProps = {
   isDisabled: false,
   customClass: null,
 };
-
-const importSampleFunction = (updateModalProps) => {
-  const title = "Import Samples from File";
+const importSampleFunction = (updateModalProps, ImportAsChemical) => {
```
Comment (FabianMauz) [inline app/packs/src/components/contextActions/ExportImportButton.js:100]: I as lowercase

**Review by FabianMauz — COMMENTED** [review]
Part I

File: app/api/helpers/report_helpers.rb [hunk 2]
```diff
@@ -192,6 +192,76 @@ def build_sql(table, columns, c_id, ids, checkedAll = false)
...
   #
+
+  def generate_sheet(table, result, columns_params, export, type)
+    case type
+    when :analyses
+      sheet_name = "#{table}_analyses".to_sym
+      export.generate_analyses_sheet_with_samples(sheet_name, result, columns_params)
+    when :chemicals
+      sheet_name = "#{table}_chemicals"
+      format_result = type == :chemicals ? Export::ExportChemicals.format_chemical_results(result) : result
```
[the remark this answers is not in the document]
Comment (adambasha0) [reply app/api/helpers/report_helpers.rb]: fixed

File: app/packs/src/components/contextActions/ExportImportButton.js (hunk 1 above)
Comment (adambasha0) [reply app/packs/src/components/contextActions/ExportImportButton.js]: fixed

**Review by FabianMauz — COMMENTED** [review]
Gone through the code together wit Adam.
Maybe add some tests for the hole usecase (import/export/report???) to ensure that all columns etc are covered by the new features.

File: lib/export/export_excel.rb [hunk 3]
```diff
@@ -11,11 +11,11 @@ def initialize(**args)
       @xfile.workbook.styles.fonts.first.name = 'Calibri'
     end
 
-    def generate_sheet_with_samples(table, samples = nil)
+    def generate_sheet_with_samples(table, samples = nil, selected_columns = nil)
```
[the remark this answers is not in the document]
Comment (adambasha0) [reply lib/export/export_excel.rb:14]: I checked it, it is needed for line 204 in  report_helper.rb

File: lib/import/import_samples.rb [hunk 4]
```diff
@@ -69,18 +66,18 @@ def extract_molfile_and_molecule(row)
     def process_row(data)
       row = [header, xlsx.row(data)].transpose.to_h
 
-      return unless structure?(row) || row['decoupled'] == 'Yes'
+      return unless structure?(row) || row['decoupled'] == 'yes'
```
[the remark this answers is not in the document]
Comment (adambasha0) [reply lib/import/import_samples.rb]: done

**Pull request merged**